GUI/AllegroInput.cpp

```diff
@@ -236,16 +236,44 @@ void AllegroInput::Update(void)
 		m_PrevNetworkMouseButtonsStates[player][0] = m_NetworkMouseButtonsEvents[player][0];
 		m_PrevNetworkMouseButtonsStates[player][1] = m_NetworkMouseButtonsEvents[player][1];
 		m_PrevNetworkMouseButtonsStates[player][2] = m_NetworkMouseButtonsEvents[player][2];
-	}
-	else 
-	{
-		m_MouseButtonsEvents[0] = mouse_b & AMBLEFT ? (m_MouseButtonsStates[0] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[0] == Down ? Released : None);
-		m_MouseButtonsEvents[1] = mouse_b & AMBMIDDLE ? (m_MouseButtonsStates[1] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[1] == Down ? Released : None);
-		m_MouseButtonsEvents[2] = mouse_b & AMBRIGHT ? (m_MouseButtonsStates[2] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[2] == Down ? Released : None);
+	} else {
+		if (!m_KeyJoyMouseCursor) {
+			if (mouse_b & AMBLEFT) {
+				m_MouseButtonsEvents[0] = (m_MouseButtonsStates[0] == Up) ? Pushed : Repeat;
+				m_MouseButtonsStates[0] = Down;
+			} else {
+				m_MouseButtonsEvents[0] = (m_MouseButtonsStates[0] == Down) ? Released : None;
+				m_MouseButtonsStates[0] = Up;
+			}
+		}
 
-		m_MouseButtonsStates[0] = mouse_b & AMBLEFT ? Down : Up;
-		m_MouseButtonsStates[1] = mouse_b & AMBMIDDLE ? Down : Up;
-		m_MouseButtonsStates[2] = mouse_b & AMBRIGHT ? Down : Up;
+		if (mouse_b & AMBMIDDLE) {
+			m_MouseButtonsEvents[1] = (m_MouseButtonsStates[1] == Up) ? Pushed : Repeat;
+			m_MouseButtonsStates[1] = Down;
+		} else {
+			m_MouseButtonsEvents[1] = (m_MouseButtonsStates[1] == Down) ? Released : None;
+			m_MouseButtonsStates[1] = Up;
+		}
+
+		if (mouse_b & AMBRIGHT) {
+			m_MouseButtonsEvents[2] = (m_MouseButtonsStates[2] == Up) ? Pushed : Repeat;
+			m_MouseButtonsStates[2] = Down;
+		} else {
+			m_MouseButtonsEvents[2] = (m_MouseButtonsStates[2] == Down) ? Released : None;
+			m_MouseButtonsStates[2] = Up;
+		}
+
+		if (m_Player < 0 || m_Player >= 4) {
+			int players = 4;
```

Comment:
I think you can just do `for (int i = 0; i < 4; i++) {` and not define the players variable here. Maybe it's nicer to make it clear what's going on though.